pkg/serverless/flush/strategy.go

```diff
@@ -0,0 +1,95 @@
+package flush
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Strategy is deciding whether the data should be flushed or not at the given moment.
+type Strategy interface {
+	String() string
+	ShouldFlush(moment Moment, t time.Time) bool
+}
+
+// Moment represents at which moment we're asking the flush strategy if we
+// should flush or not.
+// Note that there is no entry for the shutdown of the environment because we always
+// flush in this situation.
+type Moment string
+
+const (
+	// Starting is used to represent the moment the function is starting because
+	// it has been invoked.
+	Starting Moment = "starting"
+	// Stopping is used to represent the moment right after the function has finished
+	// its execution.
```

Comment:
```suggestion
	// Stopping will trigger a flush at the end of the function
```

Reply:
It's not correct, `Stopping` is used as a flag for the adaptive flush to know that it has been queried right after the function is stopping (and it won't/could not trigger a flush).